lib/Runtime/Debug/TTSerialize.cpp

```diff
@@ -963,8 +966,8 @@ namespace TTD
             return NSTokens::ParseTokenKind::Error;
         }
 
-        //convert this number to get the length of the string (not including "")
-        charList.Add(_u('\0'));
+        // Convert this number to get the length of the string (not including ""),
+        // charList is already null-terminated by the call to ScanNumber.
```

Comment:
At the end of this method, charList is recreated with no null ending. I didn't follow the whole call chain but better we add null ending there too?